base/applications/network/dwnl/dwnl.c

```diff
@@ -246,6 +246,14 @@ CBindStatusCallback_OnProgress(IBindStatusCallback *iface,
             break;
 
         case BINDSTATUS_ENDDOWNLOADDATA:
+            /* since download is completed, update progress one last time to be at 100% */
+            if(This->Size) // To avoid division by zero: if file size is known
+            {
+
+                This->Progress = This->Size; // Ensure progress == total size
```

Comment:
```suggestion
            if (This->Size) // To avoid division by zero: if file size is known
            {
                This->Progress = This->Size; // Ensure progress == total size
```

Reply:
Addressed: [dwnl-progress-fix-main 0a61b4aa0cf] Remove one extra newline